servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ProxyConnectConnectionFactoryFilter.java

```diff
@@ -86,7 +87,7 @@ public Single<C> newConnection(final ResolvedAddress resolvedAddress,
                 return delegate().newConnection(resolvedAddress, contextMap, observer).flatMap(c -> {
                     try {
                         return c.request(c.connect(connectAddress).addHeader(CONTENT_LENGTH, ZERO))
-                                .flatMap(response -> handleConnectResponse(c, response))
+                                .flatMap(response -> handleConnectResponse(c, response).shareContextOnSubscribe())
```

Comment:
for my own education: is this related to the changes below and if so why?

Reply:
no, just observed as part of the main fix. The `flatMap` is like `defer`, will make a copy of a context when it subscribes to a returned single. Because `handleConnectResponse` is 100% internal method (users can not do anything in the middle of that method) it doesn't matter if we copy the context or share. Added `shareContextOnSubscribe()` to avoid unnecessary copying, just to reduce allocations.